File: threexp1.lua
function threexp1(x)
    while x ~= 1 do
        if x % 2 == 0 then
            x = x // 2
        else
            x = 3 * x + 1
        end
        print(x)
    end
    print("OK")
end

threexp1(27)

File: threexp1.c
#include <stdio.h>

int threexp1(int x) {
    while (x != 1) {
        if (x % 2 == 0) {
            x = x / 2;
        } else {
            x = 3 * x + 1;
        }
        printf("%d\n", x);
    }
    printf("OK\n");
    return 0;
}

int main() {
    return threexp1(27);
}
